test/e2e/tests/test_table_replicas.py

```diff
@@ -0,0 +1,542 @@
+
+        # Remove the last replica
+        cr["spec"]["tableReplicas"] = [
+            {"regionName": region} for region in regions_to_keep
+        ]
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+
+        # Wait for the replica to be removed
+        table.wait_until(
+            table_name,
+            table.replicas_match(regions_to_keep),
+            timeout_seconds=REPLICA_WAIT_AFTER_SECONDS,
+            interval_seconds=30,
+        )
+
+        # Verify remaining replicas
+        replicas = table.get_replicas(table_name)
+        assert replicas is not None
+        assert len(replicas) == len(regions_to_keep)
+
+        # Check that removed region is gone and kept regions are present
+        region_names = [r["RegionName"] for r in replicas]
+        for region in regions_to_keep:
+            assert region in region_names
+        for region in regions_to_remove:
+            assert region not in region_names
+
+    def test_delete_table_with_replicas(self, table_with_replicas):
+        (ref, res) = table_with_replicas
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Delete the k8s resource
+        k8s.delete_custom_resource(ref)
+
+        max_wait_seconds = REPLICA_WAIT_AFTER_SECONDS
+        interval_seconds = 30
+        start_time = time.time()
+
+        while time.time() - start_time < max_wait_seconds:
+            if not self.table_exists(table_name):
+                break
+            time.sleep(interval_seconds)
+
+        # Verify the table was deleted
+        assert not self.table_exists(table_name)
+        replicas = table.get_replicas(table_name)
+        assert replicas is None
+
+    def test_terminal_condition_for_invalid_stream_specification(self, table_with_invalid_replicas):
+        (ref, res) = table_with_invalid_replicas
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Wait for the terminal condition to be set
+        max_wait_seconds = 120
+        interval_seconds = 10
+        start_time = time.time()
+        terminal_condition_set = False
+
+        while time.time() - start_time < max_wait_seconds:
+            cr = k8s.get_resource(ref)
+            if cr and "status" in cr and "conditions" in cr["status"]:
+                for condition_obj in cr["status"]["conditions"]:
+                    if condition_obj["type"] == "ACK.Terminal" and condition_obj["status"] == "True":
+                        terminal_condition_set = True
+                        # Verify the error message
+                        assert "table must have DynamoDB Streams enabled with StreamViewType set to NEW_AND_OLD_IMAGES" in condition_obj[
+                            "message"]
+                        break
+
+            if terminal_condition_set:
+                break
+
```

Comment:
you could also use https://github.com/aws-controllers-k8s/test-infra/blob/main/src/acktest/k8s/condition.py#L29 (i know not all controllers are using them)